File: BlogServer/Blog.Model/Param/RoutingConfigureParam.cs

using Blog.Model.Enum;

namespace Blog.Model.Param
{
    public class RoutingConfigureInsertParam
    {
        public int? Primary_id { get; set; }
        public required string Name { get; set; }
        public required string Title { get; set; }
        public required string Path { get; set; }
        public required string Component { get; set; }
        public required string IconType { get; set; }
        public required RangeEnum Range { get; set; }
        public required MenuEnum Menu { get; set; }
        public required string Redirect { get; set; }
        public required StatusEnum Status { get; set; }
        public int? SerialNumber { get; set; }
        public ContentsEnum Contents { get; set; }
    }

    public class RoutingConfigureUpdateParam : RoutingConfigureInsertParam
    { 
        public int Id { get; set; }
    }

    public class RoutingConfigureFindParam
    {
        public required RangeEnum Range { get; set; }
    }
}

File: blog-server/src/module/admin/routingConfigure/model/routingConfigure.param.ts
import {IsString, IsInt, IsOptional} from 'class-validator'


export class RoutingConfigureInsertParam {
    @IsString({ message: 'title必须是字符类型' })
    title: string

    @IsString({ message: 'path必须是字符类型' })
    path: string

    @IsString({ message: 'component' })
    component: string;

    @IsInt({ message: 'range必须是整数类型' })
    range: number;

    @IsInt({ message: 'menu必须是整数类型' })
    menu: number;

    @IsString({ message: 'redirect必须是整数类型' })
    redirect: string;

    @IsInt({ message: 'status必须是整数类型' })
    status: number;

    @IsInt({ message: 'status必须是整数类型' })
    @IsOptional()       // 表示 search 字段是可选的
    serialNumber?: number;
}


export class RoutingConfigureUpdateParam extends RoutingConfigureInsertParam{
    @IsInt({ message: 'id必须是整数类型' })
    id: number
}



export class RoutingConfigureFindParam {
    @IsInt({ message: 'range必须是整数类型' })
    range: number;
}

